### PR TITLE
ENH: #3335 Pivot table support for setting name of margins column.

### PR DESCRIPTION
**Pull request by lexual**
ref #3335.

Adds margin_column parameter to pivot_table so that user can set it to
something other than 'All'.
Raises ValueError exception if there is a conflict between the value of
margin_column and one of the other values appearing in the indices of
the pivot table.

### CODE REVIEW

File: pandas/tools/pivot.py [hunk 1]
```diff
@@ -8,13 +7,16 @@
...
 import pandas.core.common as com
 import numpy as np
 
+DEFAULT_MARGIN_COLUMN_NAME = 'All'
+
+
 def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
-                fill_value=None, margins=False, dropna=True):
+                fill_value=None, margins=False, dropna=True,
+                margins_column=DEFAULT_MARGIN_COLUMN_NAME):
```
Comment (jreback) [inline pandas/tools/pivot.py]: no need to have another variable for this, just set it here in the constructor.

File: pandas/tools/pivot.py [hunk 2]
```diff
@@ -155,28 +161,50 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
 DataFrame.pivot_table = pivot_table
 
 
-def _add_margins(table, data, values, rows, cols, aggfunc):
+def _add_margins(table, data, values, rows, cols, aggfunc,
+                 margins_column=DEFAULT_MARGIN_COLUMN_NAME):
+    exception_message = 'Must choose different value for margins_column'
```
Comment (jreback) [inline pandas/tools/pivot.py]: maybe ``conflicting name for margins``

File: pandas/tools/pivot.py [hunk 3]
```diff
@@ -155,28 +161,50 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
...
+    for level in table.index.names:
+        if margins_column in table.index.get_level_values(level):
+            raise ValueError(exception_message)
+    # could be passed a Series object with no 'columns'
```
Comment (jreback) [inline pandas/tools/pivot.py]: you can move this a bit lower (e.g. past 176)

File: pandas/tools/tests/test_pivot.py [hunk 4]
```diff
@@ -7,6 +7,7 @@
 from pandas import DataFrame, Series, Index, MultiIndex, Grouper
 from pandas.tools.merge import concat
 from pandas.tools.pivot import pivot_table, crosstab
+from pandas.tools.pivot import DEFAULT_MARGIN_COLUMN_NAME
```
Comment (jreback) [inline pandas/tools/tests/test_pivot.py]: you know its defaulted to ``All``, so just use that directly.

**Comment by jreback** [comment]
this is ok. in general, try to avoid reformatting other text unless its germane as its hard to see what you are changing. If you want to reformat, pls do that in a separate PR.

**Comment by jreback** [comment]
can you rebase / update

**Comment by jreback** [comment]
@lexual can you rebase / update

**Comment by jreback** [comment]
closing, but if you'd like to update, pls reopen

**Comment by lexual** [comment]
Hi, I've got some availability again to try and help out again.

I'll see if I can rebase this and try again.

**Comment by lexual** [comment]
Doesn't appear I can re-open though.

Maybe I have to open a brand new one ??

**Pull request closed without merging**